pandas/tests/indexes/timedeltas/test_timedelta.py

```diff
@@ -509,6 +509,10 @@ def test_fields(self):
         tm.assert_series_equal(s.dt.seconds, Series(
             [10 * 3600 + 11 * 60 + 12, np.nan], index=[0, 1]))
 
+        # preserve name (GH15589)
```

Comment:
I now ensured I have a test for each of period, timedelta, datetime that checks the name preservation, but indeed, ideally would have a test in datetimelike.py for that.